tests/sys/conn_can/main.c

```diff
@@ -145,25 +147,28 @@ static int _send(int argc, char **argv, bool rtr)
 
     if (rtr) {
         frame.can_id = CAN_RTR_FLAG | strtoul(argv[3], NULL, 16);
-        frame.can_dlc = strtoul(argv[4], NULL, 10);
+        frame.len = strtoul(argv[4], NULL, 10);
     } else {
         frame.can_id = strtoul(argv[3], NULL, 16);
-        frame.can_dlc = argc - 4;
+        frame.len = argc - 4;
     }
-    if (frame.can_dlc > 8) {
+    if (frame.len > DEFAULT_CAN_MAX_DLEN) {
```

Comment:
this check shoud probably be different for FD and non FD drivers

Reply:
yes but it is already done in `sys/include/can/common.h`:

```C
/**
 * @brief Default CAN maximum data length
 */
#ifdef MODULE_FDCAN
#define DEFAULT_CAN_MAX_DLEN CANFD_MAX_DLEN
#else
#define DEFAULT_CAN_MAX_DLEN CAN_MAX_DLEN
#endif
`